spec/light-client/Lightclient.tla

```diff
@@ -1,111 +1,273 @@
-       In the implementation, this is the oldest block,
-       where block.bftTime + trustingPeriod >= globalClock.now. *)
-    blockchain
-    (* A sequence of BlockHeaders,
-       which gives us the God's (or Daemon's) view of the blockchain. *)
-
-(* Compute the total voting power of a subset of validators Nodes,
-   whose individual voting powers are given by a function vp *)  
-RECURSIVE PowerOfSet(_, _)
-PowerOfSet(vp, Nodes) ==
-    IF Nodes = {}
-    THEN 0
-    ELSE LET node == CHOOSE n \in Nodes: TRUE IN
-        (* ASSERT(node \in DOMAIN vp) *)
-        vp[node] + PowerOfSet(vp, Nodes \ {node})
-
-(* Is the voting power correct,
-   that is, more than 2/3 of the voting power belongs to the correct processes? *)
-IsCorrectPower(vp) ==
-    LET CV == Corr \intersect DOMAIN vp
-        FV == Faulty \intersect DOMAIN vp
-        CP == PowerOfSet(vp, CV)
-        FP == PowerOfSet(vp, FV)
-    IN
-    CP > 0 /\ CP > 2 * FP
+\* the state variables of Blockchain, see Blockchain.tla for the details
+VARIABLES tooManyFaults, height, minTrustedHeight, blockchain, Faulty
+
+\* All the variables of Blockchain. For some reason, BC!vars does not work
+bcvars == <<tooManyFaults, height, minTrustedHeight, blockchain, Faulty>>
+
+(* Create an instance of Blockchain.
+   We could write EXTENDS Blockchain, but then all the constants and state variables
+   would be hidden inside the Blockchain module.
+ *) 
+BC == INSTANCE Blockchain WITH tooManyFaults <- tooManyFaults, height <- height,
+  minTrustedHeight <- minTrustedHeight, blockchain <- blockchain, Faulty <- Faulty
+
+(************************** Environment ************************************)
+NoEvent == [type |-> "None"]
+
+InEvents ==
+        (* start the client given the height to verify (fixed in our modelling) *)
+    [type: "start", heightToVerify: BC!Heights]
+       \union
+        (* receive a signed header that was requested before *)
+    [type: "responseHeader", hdr: BC!SignedHeaders]
+        \union
+    {NoEvent}
+    (* most likely, the implementation will have a timeout event, we do not need it here *)
+
+(* initially, the environment is not issuing any requests to the lite client *)
+EnvInit ==
+    inEvent = NoEvent
     
+(* The events that can be generated by the environment (reactor?):
+    user requests and node responses *)
+EnvNext ==
+    \/ /\ state = "init"
+       \* modeling feature, do not start the client before the blockchain is constructed
+       /\ height >= TO_VERIFY_HEIGHT
+       /\ inEvent' = [type |-> "start", heightToVerify |-> TO_VERIFY_HEIGHT]
+    \/ /\ state = "working"
+       /\ outEvent.type = "requestHeader"
+       /\ \E hdr \in BC!SoundSignedHeaders(outEvent.height):
+            inEvent' = [type |-> "responseHeader", hdr |-> hdr]
+
+(************************** Lite client ************************************)
+
+(* the control states of the lite client *) 
+States == { "init", "working", "finished" }
```

Comment:
We can probably reorder some definitions so it's a bit easier to understand what is the protocol being modelled, what is environment (system model somehow) and what are the properties we want to check.